src/makeCalculator.js

```diff
@@ -4,7 +4,42 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      if (typeof callback === 'function') {
+
+        callback.call(this, number);
+
+        return this;
+      }
+    },
```

Comment:
There is no error handling for the case when the `callback` is not a function. It would be safer to throw an error or return some indication that the operation was not successful.